cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseQueue.java

```diff
@@ -143,21 +144,22 @@ public Integer call(SQLDatabase db) throws Exception {
      * tasks
      */
     public void shutdown() {
-        acceptTasks = false;
-        //pass straight to queue, tasks passed via submitTaskToQueue will now be blocked.
-        queue.submit(new Runnable() {
-            @Override
-            public void run() {
-                db.close();
+        // If shutdown has already been called then we don't need to shutdown again
+        if (acceptTasks.getAndSet(false)) {
```

Comment:
Do you think we should have a warning if shutdown is called more than once? As discussed it's harmless but having a warning might help spot bad patterns in client code.